pytorch_lightning/trainer/connectors/profiler_connector.py

```diff
@@ -11,16 +11,24 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License
-from pytorch_lightning.profiler import PassThroughProfiler, SimpleProfiler
+
+from typing import Union
+
+from pytorch_lightning.profiler import BaseProfiler, PassThroughProfiler, SimpleProfiler, AdvancedProfiler
 
 
 class ProfilerConnector:
 
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def on_trainer_init(self, profiler):
+    def on_trainer_init(self, profiler: Union[BaseProfiler, bool, str]):
         # configure profiler
-        if profiler is True:
+        if profiler is True or profiler == "simple":
```

Comment:
we should not have two options to do this. IMO we should then always go with a string. Thoughts @Borda ?

Reply:
So backwards incompatibility would not be a problem?

Reply:
that's a good point. Probably we need to keep it then but deprecate it?

Reply:
Just checking:
All explicitly boolean arg values (True/False) would trigger a deprecation warning right? Also, currently when running from the CLI, one can do `my_file.py --profiler` which would be also deprecated in this case and `my_file.py --profiler simple` would be the new way of doing things.

Reply:
I feel like a string is quite sensitive to typos, so what about Enumn?

Reply:
this is not intuitive for users with argparse

Reply:
```
from enum import Enum

class Prof(Enum):
    ABC = 'abc'
    BCD = 'bcd'

print (Prof.ABC)
print (Prof('abc'))
```

Reply:
but then from the argparse it would be a string again? So still typos? I don't really think an enum is necessary here

Reply:
If using an enum it would be best to also change `--auto_scale_batch_size` which also [uses strings](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/tuner/batch_size_scaling.py#L100) and [--auto_lr_find](https://github.com/PyTorchLightning/pytorch-lightning/blob/1d7c615d82588c6b01b8926a2afd346c8cc3b8ad/pytorch_lightning/tuner/lr_finder.py#L243) as well.

Reply:
@Borda what do you think?

Reply:
> but then from the argparse it would be a string again? So still typos? I don't really think an enum is necessary here

yes but it will we caught as soon as you make that casting not later while running

> If using an enum it would be best to also change `--auto_scale_batch_size` which also [uses strings](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/tuner/batch_size_scaling.py#L100) and [--auto_lr_find](https://github.com/PyTorchLightning/pytorch-lightning/blob/1d7c615d82588c6b01b8926a2afd346c8cc3b8ad/pytorch_lightning/tuner/lr_finder.py#L243) as well.

yes, a string is ine